contracts/lib.rs

```diff
@@ -1145,64 +1143,36 @@ pub mod prosopo {
             }
             self.validate_dapp(commitment.contract)?;
 
+            let user = self.dapp_users.get(commitment.account).unwrap();
+
             // only make changes if commitment is Pending approval or disapproval
-            if commitment.status == CaptchaStatus::Pending {
-                commitment.status = CaptchaStatus::Approved;
+            if commitment.status != CaptchaStatus::Pending {
+                return Err(Error::CaptchaSolutionCommitmentNotPending);
+            }
 
-                self.record_commitment(
-                    commitment.account,
-                    captcha_solution_commitment_id,
-                    commitment,
-                );
+            self.pay_fee(&caller, &commitment.contract)?;
```

Comment:
When the payee is Dapp, the fee should not be paid if the captcha commitment is being disapproved. Otherwise, Provider funds could easily be depleted by the submission of many incorrect captcha solutions.

| Scenario | Payee | Approved | Pay Fee |
|-----|-----:|---------------|---------------|
| Data Labelling |    Dapp| False              | False             |
| Data Labelling |    Dapp| True             | True             |
| Captcha service |     Provider| False             | True            |
| Captcha service |     Provider| True             | True            |

Reply:
good point, will fix :)

Reply:
could you check the most recent commit to verify this has been fixed?